### PR TITLE
Paged api call

### PR DESCRIPTION
**Pull request by dwagg**

### CODE REVIEW

**Comment by dwagg** [comment]
We still need to decide how we want to load in the projects. The continuous method where we reload the view's list as soon as possible is currently implemented and didn't seem to cause an issue for me. It isn't hard to change it to load when the user scrolls to the bottom of the list though so that is an option as well.

File: Mobile Test Runner for Jama/Mobile Test Runner for Jama/RestHelper.swift [hunk 1]
```diff
@@ -61,10 +61,16 @@ class RestHelper {
         return request
     }
     
-    static func processRestJson(jsonData: [String: Any] ) -> [[String: AnyObject]] {
+    static func processRestJson(jsonData: [String: Any] ) -> ([[String: AnyObject]], Int) {
+        var totalItems: Int = 0
         var endpointData : [[String: AnyObject]]  = []
         var meta: [String:AnyObject] = jsonData["meta"] as! Dictionary
         let status = meta["status"] as! String
+        if let pageInfo: [String : AnyObject] = meta["pageInfo"] as? Dictionary {
```
Comment (mmcbee1) [inline Mobile Test Runner for Jama/Mobile Test Runner for Jama/RestHelper.swift]: What is this line checking for? We're assigning an immutable dictionary variable (pageInfo) to the value of a mutable dictionary variable (meta) that we had already assigned as another passed-in dictionary variable (jsonData)? And with the if statement we're checking whether pageInfo was successfully assigned in order to set the total number of Items? Could you help me understand this a little better?

File: Mobile Test Runner for Jama/Mobile Test Runner for Jama/LoginViewController.swift [hunk 2]
```diff
@@ -106,7 +106,7 @@ class LoginViewController: UIViewController, UITextFieldDelegate {
 }
 
 extension LoginViewController: EndpointDelegate{
-    func didLoadEndpoint(data: [[String : AnyObject]]?) {
+    func didLoadEndpoint(data: [[String : AnyObject]]?, totalItems: Int) {
```
Comment (benleelawrence) [inline Mobile Test Runner for Jama/Mobile Test Runner for Jama/LoginViewController.swift:109]: Delete space after String?

File: Mobile Test Runner for Jama/Mobile Test Runner for Jama/ProjectListViewController.swift [hunk 3]
```diff
@@ -57,17 +58,29 @@ class ProjectListViewController: UIViewController {
 }
 
 extension ProjectListViewController: EndpointDelegate {
-    func didLoadEndpoint(data: [[String : AnyObject]]?) {
+    func didLoadEndpoint(data: [[String : AnyObject]]?, totalItems: Int) {
```
Comment (benleelawrence) [inline Mobile Test Runner for Jama/Mobile Test Runner for Jama/ProjectListViewController.swift:61]: Delete space after String

File: Mobile Test Runner for Jama/Mobile Test Runner for Jama/RestHelper.swift [hunk 4]
```diff
@@ -61,10 +61,16 @@ class RestHelper {
         return request
     }
     
-    static func processRestJson(jsonData: [String: Any] ) -> [[String: AnyObject]] {
+    static func processRestJson(jsonData: [String: Any] ) -> ([[String: AnyObject]], Int) {
+        var totalItems: Int = 0
         var endpointData : [[String: AnyObject]]  = []
         var meta: [String:AnyObject] = jsonData["meta"] as! Dictionary
```
Comment (benleelawrence) [inline Mobile Test Runner for Jama/Mobile Test Runner for Jama/RestHelper.swift]: Add space after String:

File: Mobile Test Runner for Jama/Mobile Test Runner for Jama/RestHelper.swift [hunk 5]
```diff
@@ -61,10 +61,16 @@ class RestHelper {
         return request
     }
     
-    static func processRestJson(jsonData: [String: Any] ) -> [[String: AnyObject]] {
+    static func processRestJson(jsonData: [String: Any] ) -> ([[String: AnyObject]], Int) {
+        var totalItems: Int = 0
         var endpointData : [[String: AnyObject]]  = []
```
Comment (benleelawrence) [inline Mobile Test Runner for Jama/Mobile Test Runner for Jama/RestHelper.swift]: Delete space after endpointData

File: Mobile Test Runner for Jama/Mobile Test Runner for Jama/RestHelper.swift (hunk 1 above)
Comment (benleelawrence) [inline Mobile Test Runner for Jama/Mobile Test Runner for Jama/RestHelper.swift]: Delete space after String

File: Mobile Test Runner for Jama/Mobile Test Runner for Jama/LoginViewController.swift (hunk 2 above)
Comment (dwagg) [reply Mobile Test Runner for Jama/Mobile Test Runner for Jama/LoginViewController.swift:109]: Are you wanting to remove the space after the colon as well?

**Review by benleelawrence — APPROVED** [review]
Code formatting is consistent, tested on two iPhones and works great, ready to merge into master

**Pull request merged**